pkg/webhook/core.oam.dev/v1alpha2/applicationconfiguration/validating_handler.go

```diff
@@ -110,35 +109,6 @@ func (h *ValidatingHandler) Handle(ctx context.Context, req admission.Request) a
 	return admission.ValidationResponse(true, "")
 }
 
-// ValidateTraitObjectFn validates the ApplicationConfiguration on creation/update
-func ValidateTraitObjectFn(_ context.Context, v ValidatingAppConfig) []error {
```

Comment:
When preparing validation data, we need trait GVK to get trait definition. It should fail in the preparation phase if GVK is missed. So I move this logic into preparation phase.